data/transactions/verify/txn.go

```diff
@@ -92,6 +88,33 @@ func stxnAsyncVerify(arg interface{}) interface{} {
 }
 
 func stxnVerifyCore(s *transactions.SignedTxn, proto *config.ConsensusParams) error {
+	// Check for non-empty fields that are not yet enabled
+	if !proto.Asset {
```

Comment:
Looks good. Implementation is a bit verbosed, but the logic seems solid.
i.e.
```
if !proto.Asset && (
   (s.Txn.AssetConfigTxnFields != transactions.AssetConfigTxnFields{}) ||
   (s.Txn.AssetTransferTxnFields != transactions.AssetTransferTxnFields{}) ||
   (s.Txn.AssetFreezeTxnFields != transactions.AssetFreezeTxnFields{})) {
   return errors.New("assets not yet enabled")
}
```

Reply:
changed as suggested, thanks